target/linux/ath79/mikrotik/base-files/lib/upgrade/platform.sh

```diff
@@ -27,6 +27,22 @@ platform_do_upgrade_mikrotik_nand() {
 	nand_do_upgrade "$1"
 }
 
+platform_do_upgrade_mikrotik_nand_yafut() {
+	CI_KERNPART=none
+
+	local fw_mtd=$(find_mtd_part kernel)
+	fw_mtd="${fw_mtd/block/}"
+	[ -n "$fw_mtd" ] || return
+
+	local board_dir=$(tar tf "$1" | grep -m 1 '^sysupgrade-.*/$')
+	board_dir=${board_dir%/}
+	[ -n "$board_dir" ] || return
+
+	tar xf "$1" ${board_dir}/kernel -O | yafut -d "$fw_mtd" -w -i - -o kernel -m 0755 || echo "Kernel update failed with error $?"
```

Comment:
Do we want to bail out, and avoid doing `nand_do_update` if there is an issue with the `yafut` write?

Reply:
Right, that's a good point.  I simply mirrored what `platform_do_upgrade_mikrotik_nand()` (the `mtd erase` + `nandwrite` counterpart of this function) does and it did not seem to care a lot :-)  But yes, I think we should bail out if `yafut` reports an error.  Perhaps an even safer thing to do is to first read the kernel back from the MTD (to `/dev/null`) and only attempt writing if reading succeeds?  That should prevent messing up working devices if `yafut` gets confused for whatever reason.

Reply:
> Perhaps an even safer thing to do is to first read the kernel back from the MTD (to `/dev/null`) and only attempt writing if reading succeeds?

Answering my own question: it probably is, but if we rely on this "sanity check", then `platform_do_upgrade_mikrotik_nand()` is not going to be able to write the kernel executable to an empty Yaffs filesystem, which is rather limiting.

For that reason, I simply changed the `echo ...` part above to a `return` in 67bf7e7908330139935bfdd84b6779428d7c28b0.